bundler/lib/bundler/lockfile_parser.rb

```diff
@@ -65,18 +89,26 @@ def initialize(lockfile)
       @dependencies = {}
       @parse_method = nil
       @specs        = {}
+      @lockfile_path = begin
+        SharedHelpers.relative_lockfile_path
+      rescue GemfileNotFound
+        "Gemfile.lock"
+      end
+      @pos = Position.new(1, 1)
 
       if lockfile.match?(/<<<<<<<|=======|>>>>>>>|\|\|\|\|\|\|\|/)
-        raise LockfileError, "Your lockfile contains merge conflicts.\n" \
-          "Run `git checkout HEAD -- #{SharedHelpers.relative_lockfile_path}` first to get a clean lock."
+        raise LockfileError, "Your #{@lockfile_path} contains merge conflicts.\n" \
+          "Run `git checkout HEAD -- #{@lockfile_path}` first to get a clean lock."
       end
 
-      lockfile.split(/(?:\r?\n)+/) do |line|
+      lockfile.split(/((?:\r?\n)+)/).each_slice(2) do |line, whitespace|
```

Comment:
hmmm I think chaining the `each_slice` off of the `split` might cause the `split` to allocate an array, and the `each_slice` might (I'm not sure) allocate arrays

```
Calculating -------------------------------------
         split block   213.751k memsize (    74.015k retained)
                         1.585k objects (   701.000  retained)
                        50.000  strings (    50.000  retained)
    split each slice   243.360k memsize (    54.240k retained)
                         2.378k objects (   702.000  retained)
                        50.000  strings (    50.000  retained)

Comparison:
         split block:     213751 allocated
    split each slice:     243360 allocated - 1.14x more
Warming up --------------------------------------
         split block   207.000  i/100ms
    split each slice   183.000  i/100ms
Calculating -------------------------------------
         split block      2.072k (± 0.7%) i/s -     10.557k in   5.095838s
    split each slice      1.822k (± 0.9%) i/s -      9.150k in   5.023257s

Comparison:
         split block:     2071.8 i/s
    split each slice:     1821.7 i/s - 1.14x  slower
```

Reply:
Updated. Let me know if you thing this is better.